lgc/state/ShaderStage.cpp

```diff
@@ -96,3 +96,80 @@ const char *lgc::getShaderStageAbbreviation(ShaderStage shaderStage) {
   static const char *ShaderStageAbbrs[] = {"VS", "TCS", "TES", "GS", "FS", "CS"};
   return ShaderStageAbbrs[static_cast<unsigned>(shaderStage)];
 }
+
+// =====================================================================================================================
+// Add args to a function. The new args are put before any existing ones. This creates a new function with the
```

Comment:
So this function might finally leave a empty old function because referenced blocks are moved to the new one?

Reply:
Yes; the caller removes it. In future changes, the function being mutated might be a subfunction, and it is up to the caller of addFunctionArgs to change uses of the old function to the new function, which must be done before erasing the old function.